botocove/cove_runner.py

```diff
@@ -34,10 +34,13 @@ def __init__(
 
     def run_cove_function(self) -> CoveFunctionOutput:
         # Run decorated func with all valid sessions
-        with futures.ThreadPoolExecutor(max_workers=self.thread_workers) as executor:
+        with ThreadPoolExecutor(max_workers=self.thread_workers) as executor:
+            jobs = [executor.submit(self.cove_thread, s) for s in self.sessions]
+            outputs = (f.result() for f in as_completed(jobs))
```

Comment:
This feels like a huge nitpick comment, but I stared at this for ages trying to work out what didn't sit well with me: it's a super clean solution - I feel like `outputs` should be something that makes clear that the variable stored is a generator. 

Passing the generator directly to line 43 doesn't make it any clearer, but line 39 means you need to know exactly how this works mechanically or it's just baffling :D 

My suggestion is `completed_futures_generator`. It's not exactly a clever name :D A comment with `# https://superfastpython.com/threadpoolexecutor-in-python/#Submit_and_Use_as_Completed` feels worthwhile too.

Reply:
It's not a nitpick at all! In fact the difference between this working "smoothly" and being totally broken is exactly two characters.

If you replace the generator syntax `(...)` with list comprehension syntax `[...]` the progress bar is invisible during the whole operation until it appear at the end already 100% complete. That's because tqdm doesn't even see the input until all the futures have completed.

So I think making it more explicit is the right thing to do.

Names don't have to be clever, but they should be clear. `outputs` is about as vague as they come :-) `completed_futures_generator` does a better job of telling what it is.

I wonder if maybe a new helper function called something like `iter_each_result_as_completed` that uses the `yield` syntax would be even clearer.

Reply:
My latest commit aims to make it more explicit what's going on to implement the pattern. What do you think?

Reply:
Nice change - much more parsable imo. I've been burned enough times by `(x)` vs `(x,)` to treat braces with suspicion where possible :)